pandas/core/dtypes/cast.py

```diff
@@ -1026,3 +1027,28 @@ def find_common_type(types):
             return np.object
 
     return np.find_common_type(types, [])
+
+
+def maybe_cast_to_integer(arr, dtype):
+    """
+    Find a common data type among the given dtypes.
```

Comment:
be more explicit about what this does. It takes an *integer* dtype and returns the casted version, raising for an incompat dtype.